jccli/jc_api_v1.py

```diff
@@ -193,15 +173,20 @@ def search_systems(self, filter={}):
         query_filter = make_query_filter(filter)
 
         try:
-            api_response = self.search_api.search_systems_post(
-                content_type='application/json',
-                accept='application/json',
-                body={
-                    'filter': query_filter
-                }
-            )
-            systems = [system.to_dict() for system in api_response.results]
-            return systems
+            systems = []
+            while True:
+                api_response = self.search_api.search_systems_post(
+                    content_type='application/json',
+                    accept='application/json',
+                    body={
+                        'filter': query_filter,
+                        'limit': PAGE_LIMIT,
+                        'skip': len(systems)
+                    }
+                )
+                systems.extend([system.to_dict() for system in api_response.results])
+                if len(systems) == api_response.total_count:
+                    return systems
```

Comment:
this looks like the same code as the one in search_users, get_groups and search_systems methods.  I think it might make sense to create a common method for this.

Reply:
The technique in `get_groups()` is actually significantly different because the API doesn't seem to give a `total_count` in its response ([I'm not the only one who noticed this apparently](https://github.com/TheJumpCloud/jcapi-python/issues/25)), so I just had to have it query until it gets an empty result.

I guess I could try having the other two share a helper function.

Reply:
Do the apis that do return a total_count also just return empty responses when the 'skip' exceeds the last item? If so maybe they could be unified without using the total_count at al.

Reply:
They almost certainly do, but I still think it doesn't make sense to unify the V1 and V2 methods' pagination handling:

1. The V1 API call methods expect the `skip` and `limit` parameters to be `"key": "value"` inside the `body` argument, whereas the V2 API call methods take the `skip` and `limit` parameters as part of the method call (this is not documented AFAICT).

2. The V1 API call methods return `Systemuserlist` or `Systemlist` objects, which have iterable `.results` properties, but are not themselves iterables. Meanwhile, the V2 API call method for groups just returns a list of iterables.

3. We gain a small amount of efficiency by at least eliminating V1 API calls which would have returned no new information.

Reply:
They almost certainly do, but I still think it doesn't make sense to unify the V1 and V2 methods' pagination handling:

1. The V1 API call methods expect the `skip` and `limit` parameters to be `"key": "value"` pairs inside the `body` argument, whereas the V2 API call methods take the `skip` and `limit` parameters as part of the method call (this is not documented AFAICT).

2. The V1 API call methods return `Systemuserlist` or `Systemlist` objects, which have iterable `.results` properties, but are not themselves iterables. Meanwhile, the V2 API call method for groups just returns a list of iterables.

3. We gain a small amount of efficiency by at least eliminating V1 API calls which would have returned no new information.

So the bottom line is I think the calls are slightly heterogeneous enough that generalizing them might make the code less readable.

Reply:
thanks for looking into it and sounds good to me.